pyiron/lammps/base.py

```diff
@@ -1160,18 +1161,48 @@ def _ensure_requested_cell_deformation_allowed(self, pressure):
         if hasattr(pressure, '__len__'):
             non_diagonal_pressures = np.any([p is not None for p in pressure[3:]])
 
+            if prism is None:
+                prism = UnfoldingPrism(structure.cell)
+
             if non_diagonal_pressures:
                 try:
-                    if not self._prism.is_skewed():
-                        skew_structure = self.structure.copy()
-                        skew_structure.cell[0, 1] += 2 * self._prism.acc
-                        self.structure = skew_structure
+                    if not prism.is_skewed():
+                        skew_structure = structure.copy()
+                        skew_structure.cell[0, 1] += 2 * prism.acc
+                        return skew_structure
                 except AttributeError:
                     warnings.warn(
                         "WARNING: Setting a calculation type which uses pressure before setting the structure risks " +
                         "constraining your cell shape evolution if non-diagonal pressures are used but the structure " +
                         "is not triclinic from the start of the calculation."
                     )
+        return structure
+
+    def _validate_structure(self, pressure):
```

Comment:
This was actually the guy that got me thinking about names. I find it difficult to digest that `_validate_structure` returns `rotation_matrix`.